ckipclient/ckipclient.py

```diff
@@ -37,9 +37,20 @@ def __init__(self, ip: str, port: int, usr: str, pwd: str, safe: bool = True):
         self.usr = usr
         self.pwd = pwd
         self.safe = safe
-        self.counter = 0
 
-    def segment(self, text: str, pos: bool = True) -> list:
+    def segment(self, text: str, pos: bool = True):
+        status = False
+        results = None
+        while not status:
+            try:
+                results = self._segment(text, pos)
+                status = True
+            except ConnectionError:
```

Comment:
Changed function name to 'safe_segment'
Since the parsing code is yours, I'll leave the rest of the errors to you.